src/vision/status_quo/alan_iteratively_regresses.md

```diff
@@ -0,0 +1,90 @@
+# 😱 Status quo stories: Alan iteratively regresses performance
+
+## 🚧 Warning: Draft status 🚧
+
+This is a draft "status quo" story submitted as part of the brainstorming period. It is derived from real-life experiences of actual Rust users and is meant to reflect some of the challenges that Async Rust programmers face today. 
+
+If you would like to expand on this story, or adjust the answers to the FAQ, feel free to open a PR making edits (but keep in mind that, as they reflect peoples' experiences, status quo stories [cannot be wrong], only inaccurate). Alternatively, you may wish to [add your own status quo story][htvsq]!
+
+## The story
+
+A core part of DistriData, called DDSplit, is in charge of splitting input data records into fragments that are stored on distinct servers, and then reassembling those fragments back into records in response to user queries.
+
+DDSplit was originally implemented using Java code (plus some C, interfaced via JNI). Alan thinks that Rust could provide the same quality of service while requiring less memory, and decides to try reimplementing DDSplit in Rust.
+
+Alan develops a prototype atop tokio and the [async-trait crate][], leveraging `#[async_trait]`to be able to define `async fn` methods within traits. When Alan finishes the prototype code, he finds the prototype performance has 20% slower throughput compared to the Java version.
```

Comment:
Perhaps not super important, but where did Alan learn about `async_trait`?

Reply:
Yeah, let me spell out how that arose.

Reply:
This was super easy to write: The rustc compiler is actively directly people towards that crate today. :)  Elaborated this detail in 3bc2590641158b021e01f0a2bf65c4875eed339b